libs/librepcb/library/pkg/footprintpad.cpp

```diff
@@ -176,12 +176,43 @@ void FootprintPad::setHeight(const PositiveLength& height) noexcept {
     mRegisteredGraphicsItem->setShape(toQPainterPathPx());
 }
 
-void FootprintPad::setDrillDiameter(const UnsignedLength& diameter) noexcept {
-  mDrillDiameter = diameter;
+void FootprintPad::setDrillSize(
+    const tl::optional<DrillSize>& drillSize) noexcept {
+  mDrillSize = drillSize;
+  if (mRegisteredGraphicsItem)
+    mRegisteredGraphicsItem->setShape(toQPainterPathPx());
+}
+
+void FootprintPad::setDrillSize(const PositiveLength& width,
+                                const PositiveLength& height) noexcept {
+  mDrillSize = DrillSize(width, height);
+  if (mRegisteredGraphicsItem)
+    mRegisteredGraphicsItem->setShape(toQPainterPathPx());
+}
+
+void FootprintPad::setDrillWidth(const PositiveLength& width) {
+  // raises an exception if mDrillSize == nullopt
+  mDrillSize->setWidth(width);
```

Comment:
> In the latter case, I don't really have anything to add other than that I would not know where my code relies on exceptions thrown by tl::optional<>::operator->()

For example here :)

In this case, I would just remove all get/setDrillWidth/Height from `FootprintPad`, I would only provide `getDrillSize()` and `setDrillSize()`...